include/eve/module/core/regular/impl/simd/riscv/simd_cast.hpp

```diff
@@ -0,0 +1,183 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+namespace eve::detail
+{
+
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE wide<T, typename N::combined_type>
+                riscv_lmul_ext(wide<T, N> a) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, typename N::combined_type>>();
+  static_assert(N::combined_type::value != N::value);
+  constexpr auto in_lmul  = rvv_lmul_v<T, N>;
+  constexpr auto out_lmul = rvv_lmul_v<T, typename N::combined_type>;
+  if constexpr( out_lmul == in_lmul ) return a.storage();
+  else if constexpr( match(c, category::float32) )
+  {
+    if constexpr( out_lmul == 1 ) return __riscv_vlmul_ext_f32m1(a);
+    else static_assert(false, "[eve riscv] Unsupported");
+  }
+  else if constexpr( match(c, category::int32) )
+  {
+    if constexpr( out_lmul == 1 ) return __riscv_vlmul_ext_i32m1(a);
+    else static_assert(false, "[eve riscv] Unsupported");
+  }
+  else if constexpr( match(c, category::uint32) )
+  {
+    if constexpr( out_lmul == 1 ) return __riscv_vlmul_ext_u32m1(a);
+    else static_assert(false, "[eve riscv] Unsupported");
+  }
+  else if constexpr( match(c, category::int16) )
+  {
+    if constexpr( out_lmul == -2 ) return __riscv_vlmul_ext_i16mf2(a);
+    else if constexpr( out_lmul == 1 ) return __riscv_vlmul_ext_i16m1(a);
+    else static_assert(false, "[eve riscv] Unsupported");
+  }
+  else if constexpr( match(c, category::uint16) )
+  {
+    if constexpr( out_lmul == -2 ) return __riscv_vlmul_ext_u16mf2(a);
+    else if constexpr( out_lmul == 1 ) return __riscv_vlmul_ext_u16m1(a);
+    else static_assert(false, "[eve riscv] Unsupported");
+  }
+  else if constexpr( match(c, category::int8) )
+  {
+    if constexpr( out_lmul == -4 ) return __riscv_vlmul_ext_i8mf4(a);
+    else if constexpr( out_lmul == -2 ) return __riscv_vlmul_ext_i8mf2(a);
+    else if constexpr( out_lmul == 1 ) return __riscv_vlmul_ext_i8m1(a);
+    else static_assert(false, "[eve riscv] Unsupported");
+  }
+  else if constexpr( match(c, category::uint8) )
+  {
+    if constexpr( out_lmul == -4 ) return __riscv_vlmul_ext_u8mf4(a);
+    else if constexpr( out_lmul == -2 ) return __riscv_vlmul_ext_u8mf2(a);
+    else if constexpr( out_lmul == 1 ) return __riscv_vlmul_ext_u8m1(a);
+    else static_assert(false, "[eve riscv] Unsupported");
+  }
+  else static_assert(false, "[eve riscv] Unsupported");
+}
+
+// Logical-wide conversions.
```

Comment:
This seems like that rvv special magic to bit convert logicals to wides.

a) please leave a comment this is a special logic for rvv and that's what you are doing here.
b) I'm extremly confused why this would get called. You don't change the `enabled_for` interface to `simd_cast`. How is this called? Is it tested? https://github.com/jfalcou/eve/blob/89d8ab2092984c3511ecdefeed0ff9c4d96c4fdd/include/eve/module/core/regular/simd_cast.hpp#L70

Reply:
a) added
b) As for now testing is only `unit.arch`, we test only casting between logicals do the work good. 

Remove for now `simd_cast` support for different lmul sizes, as tests for them seems to be in different testpack.

> enabled_for

I do not need to change anything there, as there for logical already have cast between logical available:
https://github.com/jfalcou/eve/blob/main/include/eve/module/core/regular/simd_cast.hpp#L75